docs/core-semantics.md

```
@@ -0,0 +1,375 @@
+|----------|---------|
+| `and`, `&&` | Logical AND (short-circuit) |
+| `or`, `\|\|` | Logical OR (short-circuit) |
+| `not`, `!` | Logical NOT |
+
+**Invariants:**
+- Short-circuit evaluation is required
+- Truthiness: `false`, `none`, `0`, `""`, `[]` are falsy; all else truthy
+
+### Function Calls
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)